### PR TITLE
WID-247 - SpaceTimeVisualizer widget

### PR DESCRIPTION
**Pull request by peeplika**

### CODE REVIEW

File: tvbwidgets/ui/spacetime_widget.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+class SpaceTimeVisualizerWidget(TVBWidget):
+    def __init__(self, connectivity, width=600, height=400, **kwargs):
+        style = self.DEFAULT_BORDER
+        super().__init__(**kwargs, layout=style)
+        self.view_width = width
+        self.view_height = height
+        self.connectivity = connectivity
+        self.from_time = 0.00
+        self.to_time = 153.47
+        self.conduction_speed = 1.0
```
Comment (romina1601) [inline tvbwidgets/ui/spacetime_widget.py:18]: I think in the end these should be retrieved programmatically and not be hardcoded.

Comment (peeplika) [reply tvbwidgets/ui/spacetime_widget.py]: Hi, I’ve added these options—could you take a look and let me know if any improvements are needed?

File: tvbwidgets/ui/spacetime_widget.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
+import numpy as np
```
Comment (liadomide) [inline tvbwidgets/ui/spacetime_widget.py:1]: Could you pls include the header from the other files ?
You should also write a section about the `author` and put your name explicitly @peeplika

File: tvbwidgets/ui/spacetime_widget.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+<priyayadav012004@gmail.com>
+"""
+
+import numpy as np
+import pythreejs as p3
+import matplotlib.pyplot as plt
+import matplotlib.colors as mcolors
+from matplotlib.gridspec import GridSpec
+from IPython.display import display
+from tvb.datatypes.connectivity import Connectivity
```
Comment (915-Misan-Teodora) [inline tvbwidgets/ui/spacetime_widget.py:18]: You can remove the imports that are not used for now.

File: tvbwidgets/ui/spacetime_widget.py [hunk 4]
```diff
@@ -169,48 +174,63 @@ def _create_matplotlib_graphs(self):
...
 
     def _add_options(self):
-        self.options = HBox(layout = Layout(width = '500px'))
+        self.options = HBox(layout = Layout(width = '800px'))
+        conduction_speed = self.connectivity.speed
+        max_time = self.connectivity.tract_lengths.max() / conduction_speed
+        min_time = self.connectivity.tract_lengths.min() / conduction_speed
         self.option_conduction_speed = FloatText(
-                value=1.0,
+                value=conduction_speed,
```
Comment (romina1601) [inline tvbwidgets/ui/spacetime_widget.py:182]: Sorry, I know I said take this from the connectivity, but I think this should actually be 1 when first rendering the plots 😅 please change it back

**Comment by liadomide** [comment]
Few suggestions to make this widget look more complete:
- add a border around all the gui components (recommended `TVBWidget.DEFAULT_BORDER`)
- rename `SpaceTimeVisualizerWidget.ipynb` into `ConnectivityStaceTimeWidget`
- in the .ipynb add a title and few word on its purpose (for example `Display a large scale Connectivity split into time slices, based in the delays of passing the signals to adjacent regions`; maybe briefly write credits to GSOC program ?)
- make the "interactive Viewer" by default as large as the "plots overview")
- in the interactive viewer it is not very easy for me to see which slice I am to select? Could we maybe leave more space between slices? The first slice is different (it shows the entire matrix) maybe we could emphasise that aspect (e.g. more space next to it, a different border?)
- "Plot details" currently move the entire screen when they are expanded, and if i change the cond-speed while the Plot details are collapsed, these details become immediately visible. Maybe we could make the details always visible (or we fix the above issues) ?

**Review by liadomide — APPROVED** [review]
This looks good to me.
@915-Misan-Teodora in what order do you advice us to merge the PRs on this repo?
Can we merge this first ?

**Pull request merged**